server/src/main/java/org/elasticsearch/search/aggregations/AggregatorBase.java

```diff
@@ -124,6 +125,26 @@ public final Function<byte[], Number> pointReaderIfAvailable(ValuesSourceConfig
         return config.getPointReaderOrNull();
     }
 
+    /**
+     * The same as {@link AggregatorBase#pointReaderIfAvailable(ValuesSourceConfig)} but allows the point reader
+     * within sampling contexts. The sampler will do its best to skip reading documents and allow aggs to calculate via the point reader.
+     *
+     * This is fine as long as scaling isn't required as combining sampled and non-sampled shards could introduce bias
+     * if scaling is not handled carefully
+     *
+     * @param config The config for the values source metric
+     * @return the point reader
+     */
+    public final Function<byte[], Number> pointReaderIfAvailableAndSamplingScalingNotRequired(ValuesSourceConfig config) {
```

Comment:
@not-napoleon I suppose? I was just thinking of plugins that may use that other function

Reply:
I really don't like having two functions that do almost the same thing, with a slightly different predicate.  It's difficult to know which one is appropriate to use where.

You're saying you want two nearly identical functions because there might be a plugin that has added a points-optimized aggregation, which also supports sampling, but cannot use points under sampling.  Do you know of such a thing?  Because if we don't have a concrete case, I'd rather err on the side of a simpler API.  Plugin authors are still free to add a check beyond this one.